plugin/src/software/aws/toolkits/eclipse/amazonq/lsp/editor/ActiveEditorChangeListener.java

```diff
@@ -0,0 +1,210 @@
+        this.languageServer = languageServer;
+        this.executor = executor;
+    }
+
+    @Override
+    public void partActivated(final IWorkbenchPartReference partRef) {
+        if (partRef.getPart(false) instanceof ITextEditor) {
+            handleEditorChange((ITextEditor) partRef.getPart(false));
+        }
+    }
+
+    @Override
+    public void partBroughtToTop(final IWorkbenchPartReference partRef) {
+        if (partRef.getPart(false) instanceof ITextEditor) {
+            handleEditorChange((ITextEditor) partRef.getPart(false));
+        }
+    }
+
+    @Override
+    public void partClosed(final IWorkbenchPartReference partRef) {
+        // When editor is closed, send notification with null values
+        handleEditorChange(null);
+    }
+
+    @Override
+    public void partDeactivated(final IWorkbenchPartReference partRef) {
+        // No action needed
+    }
+
+    @Override
+    public void partOpened(final IWorkbenchPartReference partRef) {
+        if (partRef.getPart(false) instanceof ITextEditor) {
+            handleEditorChange((ITextEditor) partRef.getPart(false));
+        }
+    }
+
+    @Override
+    public void partHidden(final IWorkbenchPartReference partRef) {
+        // No action needed
+    }
+
+    @Override
+    public void partVisible(final IWorkbenchPartReference partRef) {
+        if (partRef.getPart(false) instanceof ITextEditor) {
+            handleEditorChange((ITextEditor) partRef.getPart(false));
+        }
+    }
+
+    @Override
+    public void partInputChanged(final IWorkbenchPartReference partRef) {
+        if (partRef.getPart(false) instanceof ITextEditor) {
+            handleEditorChange((ITextEditor) partRef.getPart(false));
+        }
+    }
+
+    private void handleEditorChange(final ITextEditor editor) {
+        // Cancel any pending notification
+        if (debounceTask != null) {
+            debounceTask.cancel(false);
+        }
+
+        // Schedule a new notification after the debounce period
+        debounceTask = executor.schedule(() -> {
+            try {
+                Map<String, Object> params = createActiveEditorParams(editor);
+
+                // Send notification to language server
+                languageServer.activeEditorChanged(params);
+                Activator.getLogger().info("Active editor changed notification sent: "
+                    + (editor != null ? editor.getTitle() : "no editor"));
+
+            } catch (Exception e) {
+                Activator.getLogger().error("Failed to send active editor changed notification", e);
+            }
+        }, DEBOUNCE_DELAY_MS, TimeUnit.MILLISECONDS);
+    }
+
+    private Map<String, Object> createActiveEditorParams(final ITextEditor editor) {
+        Map<String, Object> params = new HashMap<>();
+
+        if (editor != null) {
+            IEditorInput editorInput = editor.getEditorInput();
```

Comment:
there is already logic in QEclipseEditorUtils class to get this information(entirety of this function's logic). Please reuse that instead of creating a new one. We also have logic in addEditorState in ChatCommunicationManager that does something very similar. Please reference these two